### PR TITLE
Metrics router and atracker:  Updated platform-services-go-sdk to fetch Madrid endpoint

### PR DESCRIPTION
**Pull request by adya1303**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by adya1303** [comment]
Ran Metrics-router test - 

make fmt && make testacc TEST=./ibm/service/metricsrouter TESTARGS='-run=TestAccIBMMetricsRouter.*'
gofmt -w $(find .  -path ./.direnv -prune -false -o -name '*.go' |grep -v vendor)
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/metricsrouter -v -run=TestAccIBMMetricsRouter.* -timeout 700m
=== RUN   TestAccIBMMetricsRouterRoutesDataSourceBasic
--- PASS: TestAccIBMMetricsRouterRoutesDataSourceBasic (34.29s)
=== RUN   TestAccIBMMetricsRouterTargetsDataSourceBasic
--- PASS: TestAccIBMMetricsRouterTargetsDataSourceBasic (32.38s)
=== RUN   TestAccIBMMetricsRouterTargetsDataSourceAllArgs
--- PASS: TestAccIBMMetricsRouterTargetsDataSourceAllArgs (25.58s)
=== RUN   TestAccIBMMetricsRouterRouteBasic
--- PASS: TestAccIBMMetricsRouterRouteBasic (39.43s)
=== RUN   TestAccIBMMetricsRouterRouteConfigBasicWithoutAction
--- PASS: TestAccIBMMetricsRouterRouteConfigBasicWithoutAction (25.77s)
=== RUN   TestAccIBMMetricsRouterRouteSendEmptyTarget
--- PASS: TestAccIBMMetricsRouterRouteSendEmptyTarget (0.36s)
=== RUN   TestAccIBMMetricsRouterRouteDropEmptyTarget
--- PASS: TestAccIBMMetricsRouterRouteDropEmptyTarget (0.36s)
=== RUN   TestAccIBMMetricsRouterRouteDropNoTarget
--- PASS: TestAccIBMMetricsRouterRouteDropNoTarget (22.26s)
=== RUN   TestAccIBMMetricsRouterRouteSendNoTarget
--- PASS: TestAccIBMMetricsRouterRouteSendNoTarget (8.73s)
=== RUN   TestAccIBMMetricsRouterSettingsBasic
--- PASS: TestAccIBMMetricsRouterSettingsBasic (27.71s)
=== RUN   TestAccIBMMetricsRouterSettingsAllArgs
--- PASS: TestAccIBMMetricsRouterSettingsAllArgs (25.67s)
=== RUN   TestAccIBMMetricsRouterSettingsEmptyTarget
--- PASS: TestAccIBMMetricsRouterSettingsEmptyTarget (0.36s)
=== RUN   TestAccIBMMetricsRouterTargetBasic
--- PASS: TestAccIBMMetricsRouterTargetBasic (42.96s)
=== RUN   TestAccIBMMetricsRouterTargetAllArgs
--- PASS: TestAccIBMMetricsRouterTargetAllArgs (40.56s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/metricsrouter	329.005s

**Comment by adya1303** [comment]
Ran atracker tests - 

make fmt && make testacc TEST=./ibm/service/atracker TESTARGS='-run=TestAccIBMAtracker.*'
gofmt -w $(find .  -path ./.direnv -prune -false -o -name '*.go' |grep -v vendor)
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/atracker -v -run=TestAccIBMAtracker.* -timeout 700m

=== RUN   TestAccIBMAtrackerRoutesDataSourceBasic
--- PASS: TestAccIBMAtrackerRoutesDataSourceBasic (31.05s)
=== RUN   TestAccIBMAtrackerTargetsDataSourceBasic
--- PASS: TestAccIBMAtrackerTargetsDataSourceBasic (26.39s)
=== RUN   TestAccIBMAtrackerTargetsDataSourceAllArgs
--- PASS: TestAccIBMAtrackerTargetsDataSourceAllArgs (24.76s)
=== RUN   TestAccIBMAtrackerRouteBasic
--- PASS: TestAccIBMAtrackerRouteBasic (47.30s)
=== RUN   TestAccIBMAtrackerRouteBasicMultipleRules
--- PASS: TestAccIBMAtrackerRouteBasicMultipleRules (45.39s)
=== RUN   TestAccIBMAtrackerSettingsBasic
--- PASS: TestAccIBMAtrackerSettingsBasic (42.17s)
=== RUN   TestAccIBMAtrackerTargetBasic
--- PASS: TestAccIBMAtrackerTargetBasic (44.89s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/atracker	263.819s

File: examples/ibm-metrics-router/variables.tf [hunk 1]
```diff
@@ -10,7 +10,7 @@ variable "metrics_router_target_name" {
   default     = "my-mr-target"
 }
 variable "metrics_router_target_destination_crn" {
-  description = "The CRN of a destination service instance or resource."
+  description = "The CRN of a destination service instance or resource. Ensure you have a service authorization between IBM Cloud Metrics Routing and your Cloud resource. Read [S2S authorization](https://cloud.ibm.com/docs/metrics-router?topic=metrics-router-target-monitoring&interface=ui#target-monitoring-ui) for details."
```
Comment (johnymathew) [inline examples/ibm-metrics-router/variables.tf:13]: description = "The CRN of a destination sysdig-monitor instance. Ensure you have a service authorization between IBM Cloud Metrics Routing and your sysdig monitor instance. ....

somehow convey it is the sysdig instance crn we want.

File: ibm/service/metricsrouter/data_source_ibm_metrics_router_targets.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func DataSourceIBMMetricsRouterTargets() *schema.Resource {
 						"destination_crn": &schema.Schema{
 							Type:        schema.TypeString,
 							Computed:    true,
-							Description: "The CRN of the destination service instance or resource.",
+							Description: "The CRN of the destination service instance or resource. Ensure you have a service authorization between IBM Cloud Metrics Routing and your Cloud resource. Read [S2S authorization](https://cloud.ibm.com/docs/metrics-router?topic=metrics-router-target-monitoring&interface=ui#target-monitoring-ui) for details.",
```
Comment (johnymathew) [inline ibm/service/metricsrouter/data_source_ibm_metrics_router_targets.go:53]: same as above

File: ibm/service/metricsrouter/resource_ibm_metrics_router_target.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ func ResourceIBMMetricsRouterTarget() *schema.Resource {
 				Type:         schema.TypeString,
 				Required:     true,
 				ValidateFunc: validate.InvokeValidator("ibm_metrics_router_target", "destination_crn"),
-				Description:  "The CRN of a destination service instance or resource.",
+				Description:  "The CRN of a destination service instance or resource. Ensure you have a service authorization between IBM Cloud Metrics Routing and your Cloud resource. Read [S2S authorization](https://cloud.ibm.com/docs/metrics-router?topic=metrics-router-target-monitoring&interface=ui#target-monitoring-ui) for details.",
```
Comment (johnymathew) [inline ibm/service/metricsrouter/resource_ibm_metrics_router_target.go:39]: same

File: website/docs/d/metrics_router_targets.html.markdown [hunk 4]
```diff
@@ -34,7 +34,7 @@ In addition to all argument references listed, you can access the following attr
 Nested scheme for **targets**:
 	* `created_at` - (String) The timestamp of the target creation time.
 	* `crn` - (String) The crn of the target resource.
-	* `destination_crn` - (String) The CRN of the destination service instance or resource.
+	* `destination_crn` - (String) The CRN of the destination service instance or resource. Ensure you have a service authorization between IBM Cloud Metrics Routing and your Cloud resource. Read [S2S authorization](https://cloud.ibm.com/docs/metrics-router?topic=metrics-router-target-monitoring&interface=ui#target-monitoring-ui) for details.
```
Comment (johnymathew) [inline website/docs/d/metrics_router_targets.html.markdown:37]: same

**Review by johnymathew — APPROVED** [review]

**Review by hkantare — APPROVED** [review]

**Pull request merged**